### DISCUSSION TITLE
sqlite3: attempt to write a readonly database

### DISCUSSION THREAD

**Discussion by rafaelnp** [gogs/gogs#6982] [Q&A]
I installed gogs 0.12.7-1 on Arch Linux and I am getting the following error:

`Database setting is not correct: attempt to write a readonly database`

I fllowed the documentation from [Arch Linux Wiki](https://wiki.archlinux.org/title/Gogs) and:
* started gogs as a systemd service
* enter the full path for the database: /var/lib/gogs/data/gogs.db
* created an empty data empty database in the path above using the command `sqlite3 gogs.db "VACUUM;"`

IT looks like it is not the right thing to do, however after reading [gogs documentation](https://gogs.io/docs), I could not find any additional information.  Did I miss something? Thanks in advance.

Cheers,
Rafael.

**Comment by unknwon** [comment]
This looks to me that the user that runs Gogs does not have write access to `/var/lib/gogs/data/gogs.db` (created by another user when you used `sqlite3 gogs.db "VACUUM;"`). Can you double check on that?

You don't need to manually create SQLite3 database file, it is created for you if previously didn't exist.

**Comment by rafaelnp** [comment]
I deleted the file and checked the permissions. All directories and files under `/var/lib/gogs` belong to `gogs:gogs` user and group.  I tried afterwards and got this error. It looks like something it still missing or not correctly configured

![image](https://user-images.githubusercontent.com/1737706/170979123-79bc4727-9aaa-4e60-a94c-314eaa8c398b.png)

**Reply by rafaelnp** [reply]
I checked the permissions and `/var/lib/gogs/data` has wrong permissions. I fixed them and re-ran the install process and I got an user error message.  I started gogs using `systemctl start gogs.service`, so I do not understand why the user git is being used.

![image](https://user-images.githubusercontent.com/1737706/170980705-c97ffad2-d646-469b-ab9c-d6bf10ce6a85.png)

**Reply by unknwon** [reply]
~~Hmm.. is the directory "/var/lib/gogs/data" exists?~~

~~Also just to make sure we are on the same page, which deployment method you're using? Binary or Docker?~~

**Reply by unknwon** [reply]
By convention and to prevent mistakes, Gogs expects you to run as the user "git", but you can change it to "gogs" in your case:

https://github.com/gogs/gogs/blob/519aeefbd93adad833cb45ba36f71622d3068223/conf/app.ini#L9

**Reply by rafaelnp** [reply]
OK, thanks for the explanation. If I understood correctly, what the Arch Linux Wiki describes as an installation and configuration procedure for gogs is not a good practice, because it uses the gogs user and group as the user to own all gogs files. Would it be better/advisable to install from a binary?

**Reply by unknwon** [reply]
I don't think which is necessarily better than the other, but if you ask me, I'd personally recommend binary or Docker. With binary installation though, you still need to manage to craft your systemd for running it in the background, or just nohup/screen.
